org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/entitytemplates/artifacttemplates/ArtifactTemplateResource.java

```diff
@@ -86,6 +92,30 @@ public void synchronizeReferences() throws IOException {
 		BackendUtils.synchronizeReferences((ArtifactTemplateId) this.id);
 	}
 
+	@POST
+	@Consumes(MediaType.APPLICATION_JSON)
+	public Response copySourceToFilesResource(ArtifactResourcesApiData data) {
+		List<String> artifactList = data.getArtifactList();
+		ArtifactTemplateDirectoryId sourceDir = new ArtifactTemplateSourceDirectoryId((ArtifactTemplateId) this.id);
+		FilesResource filesResource = getFilesResource();
+		for (RepositoryFileReference ref : RepositoryFactory.getRepository().getContainedFiles(sourceDir)) {
+			if (artifactList == null || artifactList.contains(ref.getFileName())) {
+				try {
+					InputStream inputStream = RepositoryFactory.getRepository().newInputStream(ref);
+					String fileName = ref.getFileName();
+					String subDirectory = "";
+					if (ref.getSubDirectory().isPresent()) {
+						subDirectory = ref.getSubDirectory().get().toString();
+					}
+					filesResource.putFile(fileName, subDirectory, inputStream);
+				} catch (IOException e) {
+					return Response.noContent().build();
```

Comment:
Please `debug` log the exception. Further, return an internal server error. The user should be notified that an error happened.